hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -181,8 +200,6 @@ private[sql] class AvroDeserializer(rootAvroType: Schema,
           case b: ByteBuffer =>
             val bytes = new Array[Byte](b.remaining)
             b.get(bytes)
-            // Do not forget to reset the position
-            b.rewind()
```

Comment:
why remove? this was a recent fix, right?

Reply:
Good point, this was reverted by mistake. I've added it back